Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml

```diff
@@ -50,3 +53,33 @@
   - type: WarpPoint
     follow: true
     location: singularity
+  - type: Sprite
+    sprite: Structures/Power/Generation/Singularity/singularity_1.rsi
+    layers:
+    - map: [ "VisualLevel" ]
+      state: singularity_1
+  - type: GenericVisualizer
+    visuals:
+      enum.SingularityAppearanceKeys.Singularity:
+       VisualLevel:
+          1:
+            sprite: Structures/Power/Generation/Singularity/singularity_1.rsi
+            state: singularity_1
+          2:
+            sprite: Structures/Power/Generation/Singularity/singularity_2.rsi
+            state: singularity_2
+          3:
+            sprite: Structures/Power/Generation/Singularity/singularity_3.rsi
+            state: singularity_3
+          4:
+            sprite: Structures/Power/Generation/Singularity/singularity_4.rsi
+            state: singularity_4
+          5:
+            sprite: Structures/Power/Generation/Singularity/singularity_5.rsi
+            state: singularity_5
+            scale: 1.5,1.5
```

Comment:
if you change the scale at level 5 & 6, you need to make sure to set it back to 1,1 for the other levels.